design/ceph/object/store.md

```diff
@@ -221,6 +221,292 @@ spec:
+  namespace: rook-ceph
+spec:
+  ...
+```
+
+If needed, create additional pool to use it in custom StorageClass, for example `GLACIER` and `REDUCED_REDUNDANCY`:
+> [!NOTE]
+> Ceph allows arbitrary name for StorageClasses, however some clients/libs insist on AWS names so it is recommended to use one of the [valid x-amz-storage-class values](https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObject.html#API_PutObject_RequestSyntax) for better compatibility:
+>
+> `STANDARD | REDUCED_REDUNDANCY | STANDARD_IA | ONEZONE_IA | INTELLIGENT_TIERING | GLACIER | DEEP_ARCHIVE | OUTPOSTS | GLACIER_IR | SNOW | EXPRESS_ONEZONE`
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: glacier-data-pool
+  namespace: rook-ceph
+spec:
+  ...
+---
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: reduced-data-pool
+  namespace: rook-ceph
+spec:
+  ...
+```
+
+Then update `CephObjectStore` CRD  `sharedPools` section with target placements and StorageClasses definitions:
+
+```yaml
+spec:
+  # [OPTIONAL] - sharedPools is mutually exclusive with spec.metadataPool and spec.dataPool
+  # It is the place to tell RGW which pools can be used to store buckets and objects data and metadata.
+  # All pools under this section should be created beforehand.
+  # Rook will append rados namespace to each of the pools, so it is safe to use the same pool across multiple RGW instances.
+  # It is also safe to use the same pool in different target placements.
+  sharedPools:
+    #  [EXISTING,OPTIONAL] - metadata pool. Will be used by default to store bucket index
+    # default poolPlacement.metadataPoolName will override this value, so it is not makes sense to use together with poolPlacements.
+    metadataPoolName: rgw-meta-pool # results to rgw-meta-pool:<CephObjectStore.name>.index
+
+    #  [EXISTING,OPTIONAL] - data pool. Will be used by default to store bucket objects
+    # default poolPlacement.dataPoolName will override this value, so it is not makes sense to use together with poolPlacements.
+    dataPoolName: rgw-data-pool # results to rgw-data-pool:<CephObjectStore.name>.data
+
+    # New settings:
+    # [OPTIONAL] - defines pool placement targets.
+    # Placement targets control which Pools are associated with a particular bucket.
+    # See: https://docs.ceph.com/en/latest/radosgw/placement/#placement-targets
+    poolPlacements:
+      # Default pool placement:
+      - name: "default"                         # [REQUIRED] - placement id. Placement with name "default" will be used as default,
+                                                # and therefore will override sharedPools.metadataPoolName and sharedPools.dataPoolName if presented.
+                                                # This means that default poolPlacement is also mutually exclusive with spec.metadataPool and spec.dataPool
+        metadataPoolName: rgw-meta-pool         # [REQUIRED] results to rgw-meta-pool:<CephObjectStore.name>.index
```

Comment:
Should we disallow `metadataPoolName`, `dataPoolName`, and `dataNonECPoolName` for the "default" placement? If they are allowed, the `sharedPools` pool names will just be ignored, right? Perhaps it is less confusing to disallow them, instead of the user trying to understand the overriding.

Reply:
i think that the problem is that `sharedPools` sets default placement to json in Ceph but it is not reflected in its name (`sharedPools.metadataPool` and `sharedPools.dataPool`). And since we cannot rename it to something like `sharedPools.defaultMetadataPool` we have 2 options:
1. leave `sharedPools` like it is and advise users to use `poolPlacements` instead. In this case it is probably ok to override.
2. remove "default" placement from `poolPlacements` and add comment to use `sharedPools.metadataPool` as default placement. In this case we should add `sharedPools.dataNonECPoolName` and `sharedPools.storageClasses` to `sharedPools` root level to be consistent.

In my opinion first approach is more straightforward but i am not an expert. Is it makes sense?

Reply:
Ok, let's go with the first option then and just document the override behavior.

Reply:
fixed:
- renamed placements to `cold` and `fast` (took naming from [RH docs](https://docs.redhat.com/en/documentation/red_hat_ceph_storage/3/html/ceph_object_gateway_for_production/assembly-developing-storage-strategies-rgw-adv#data-placement-summary-rgw-adv))
- removed class definition from all placements except `cold` to show that storageClass is optional
- added comments about `STANDARD` storage class
- used `GLACIER` storage class only for `cold` placement

Reply:
@travisn is something else missing from design doc?